dotCMS/src/main/java/com/dotmarketing/db/DotRunnableThread.java

```diff
@@ -4,63 +4,84 @@
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 import com.dotmarketing.business.DotStateException;
-import com.dotmarketing.exception.DotDataException;
 import com.dotmarketing.portlets.contentlet.model.Contentlet;
 import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
 
 public class DotRunnableThread extends Thread {
 
-    final List<DotRunnable> listeners;
-
-    public DotRunnableThread(final List<DotRunnable> listeners) {
-
-        this.listeners = listeners;
-    }
-
+  private final List<DotRunnable> listeners;
+  private final List<DotRunnable> flushers;
+  private final Thread networkCacheFlushThread = new Thread("NetworkCacheFlushThread") {
     @Override
     public void run() {
-        try {
-            LocalTransaction.wrap(this::internalRunner);
-        } catch (Exception dde) {
-            throw new DotStateException(dde);
-        }
+      try {
+        Thread.sleep(Config.getLongProperty("NETWORK_CACHE_FLUSH_DELAY", 3000));
+      } catch (InterruptedException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)